icaruscode/PMT/OpReco/ICARUSFlashAssAna_module.cc

```diff
@@ -107,23 +131,37 @@ class opana::ICARUSFlashAssAna : public art::EDAnalyzer {
   private:
 
     art::InputTag fTriggerLabel;
-    std::vector<art::InputTag> fOpHitLabel;
+    bool fSaveWaveformInfo;
+    std::vector<art::InputTag> fOpDetWaveformLabels;
+    std::vector<art::InputTag> fOpHitLabels;
     std::vector<art::InputTag> fFlashLabels;
     float fPEOpHitThreshold;
+    bool fDebug;
 
 
     TTree *fEventTree;
+    std::vector<TTree*> fOpDetWaveformTrees;
     std::vector<TTree*> fOpFlashTrees;
     std::vector<TTree*> fOpHitTrees;
+    std::vector<TTree*> fOpHitFlashTrees;
 
     int m_run;
     int m_event;
     int m_timestamp;
     //int m_nflashes;
     //int m_nophit;
+    short m_baseline;
+    short m_chargesum;
+    int m_nticks;
     float m_beam_gate_start=-99999;
     float m_beam_gate_width=-99999;
     int m_beam_type=-1;
+    unsigned int m_gate_type;
+    std::string m_gate_name;
+    int m_trigger_timestamp;
```

Comment:
The trigger timestamp from [`sbn::ExtraTriggerInfo::triggerTimestamp`](https://icarus-exp.fnal.gov/at_work/software/doc/icaruscode/latest/structsbn_1_1ExtraTriggerInfo.html#acbbec1e38c4afbcc587f2c2332668dd0) is absolute in nanoseconds from The Epoch. The only data type able to contain all that information is `std::uint64_t`.